src/plugins/console/public/services/autocomplete.ts

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { createGetterSetter } from '@kbn/kibana-utils-plugin/public';
+import type { HttpSetup } from '@kbn/core/public';
+import type { MappingsApiResponse } from '../lib/autocomplete_entities/types';
+import { API_BASE_PATH } from '../../common/constants';
+import {
+  Alias,
+  DataStream,
+  Mapping,
+  LegacyTemplate,
+  IndexTemplate,
+  ComponentTemplate,
+} from '../lib/autocomplete_entities';
+import { DevToolsSettings, Settings } from './settings';
+
+export class AutocompleteInfo {
+  public readonly alias = new Alias();
+  public readonly mapping = new Mapping();
+  public readonly dataStream = new DataStream();
+  public readonly legacyTemplate = new LegacyTemplate();
+  public readonly indexTemplate = new IndexTemplate();
+  public readonly componentTemplate = new ComponentTemplate();
+  private http!: HttpSetup;
+  private pollTimeoutId: ReturnType<typeof setTimeout> | undefined;
+
+  public setup(http: HttpSetup) {
+    this.http = http;
+  }
+
+  public getFactoryFor(
```

Comment:
Nit: I associate the term "factory" with the [Factory Pattern](https://en.wikipedia.org/wiki/Factory_method_pattern) and I'm using to thinking of factories as methods or classes that return instances of other classes. In this case, we're creating a function that's used for retrieving autocomplete entities. The complement to "retrieving" is "providing". So I suggest changing this name to `getEntityProvider`.